### PR TITLE
feat(pci-private-network): add the delete modal/action for global reg…

### PR DESCRIPTION
**Pull request by seven-amid**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/pci-private-network
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes/no
| Tickets          | DTCORE-2080 & DTCORE-2078
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+  TAggregatedNetwork,
+} from '@/api/data/network';
 import { TGateway, TRegion } from '@/api/data/project';
 import { getSubnets, TSubnet } from '@/api/data/subnets';
 import {
   getLocalZoneRegions,
   isLocalZoneRegion,
   paginateResults,
 } from '@/api/utils/utils';
+import queryClient from '@/queryClient';
```
Comment (chipp972) [inline packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts:17]: Don't import queryClient, use the `useQueryClient` hook
You will create a new instance, not use the one in the context

File: packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts [hunk 2]
```diff
@@ -220,3 +228,37 @@ export const useLocalZoneNetworks = (
...
+        queryKey: getLocalZoneRegionsQueryKey(projectId),
+      });
+      return onSuccess();
+    },
+  });
+
+  return {
+    deleteNetwork: () => {
+      return mutation.mutate();
+    },
```
Comment (chipp972) [inline packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts:261]:
```suggestion
    deleteNetwork: mutation.mutate,
```

File: packages/manager/apps/pci-private-network/src/components/delete/DeleteModal.component.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        variant={ODS_BUTTON_VARIANT.ghost}
+        onClick={onClose}
+      >
+        {t('pci_projects_project_network_private_delete_cancel')}
+      </OsdsButton>
+      <OsdsButton
+        slot="actions"
+        color={ODS_THEME_COLOR_INTENT.primary}
+        onClick={onConfirm}
+        {...(isPending ? { disabled: true } : {})}
```
Comment (chipp972) [inline packages/manager/apps/pci-private-network/src/components/delete/DeleteModal.component.tsx:68]:
```suggestion
        disabled={isPending || undefined}
```

Shorter syntax

File: packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts (hunk 1 above)
Comment (seven-amid) [reply packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts:17]: this one is imported from a file defining one instance client 

```
import { QueryClient } from '@tanstack/react-query';

export const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 300_000,
    },
  },
});

export default queryClient;
```

Comment (chipp972) [reply packages/manager/apps/pci-private-network/src/api/hooks/useNetwork.ts:17]: Ok but still not a good practice
Take it from the context, there is a hook to do that

**Review by chipp972 — APPROVED** [review]

**Review by kqesar — APPROVED** [review]

**Pull request merged**